src/app/clusters/time-synchronization-server/time-synchronization-cluster.h

```diff
@@ -123,33 +105,90 @@ class TimeSynchronizationServer : public FabricTable::Delegate
     // ReadClient::Callback functions
     void OnAttributeData(const ConcreteDataAttributePath & aPath, TLV::TLVReader * apData, const StatusIB & aStatus) override;
     void OnDone(ReadClient * apReadClient) override;
-
 #endif
 
-    CHIP_ERROR AttemptToGetTimeFromTrustedNode();
-
     // Platform event handler functions
     void OnPlatformEventFn(const DeviceLayer::ChipDeviceEvent & event);
-
-    void OnTimeSyncCompletionFn(TimeSourceEnum timeSource, GranularityEnum granularity);
+    void OnTimeSyncCompletionFn(TimeSynchronization::TimeSourceEnum timeSource, TimeSynchronization::GranularityEnum granularity);
     void OnFallbackNTPCompletionFn(bool timeSyncSuccessful);
 
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
+
+    std::optional<DataModel::ActionReturnStatus> InvokeCommand(const DataModel::InvokeRequest & request,
+                                                               TLV::TLVReader & input_arguments, CommandHandler * handler) override;
+
+    CHIP_ERROR AcceptedCommands(const ConcreteClusterPath & path,
+                                ReadOnlyBufferBuilder<DataModel::AcceptedCommandEntry> & builder) override;
+
+    CHIP_ERROR GeneratedCommands(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<CommandId> & builder) override;
+
+    const TimeSynchronization::GranularityEnum & GetGranularity() const { return mGranularity; }
+
+    TimeSynchronization::Delegate * GetDelegate() const { return mDelegate; }
+    void SetDelegate(TimeSynchronization::Delegate * delegate)
+    {
+        if (delegate != nullptr)
```

Comment:
This was existing code ... but it is crazy: you cannot say that a delegate exceeded its lifetime :( I wonder if we should just VerifyOrDie that delegate is not null. Silently ignoring nulls seems off.